src/sc_client/constants/sc_type/sc_type.py

```diff
@@ -0,0 +1,198 @@
+
+import warnings
+
+from sc_client.constants.exceptions import InvalidTypeError
+from sc_client.constants.sc_type import bitmasks
+
+
+class ScType:
+    def __init__(self, value: int = 0):
+        if not isinstance(value, (ScType, int)):
+            raise InvalidTypeError("You should to use int or ScType type for ScType initialization")
+        if isinstance(value, ScType):
+            value = value.value
+        self.value = value
+
+    def __repr__(self) -> str:
+        return f"ScType({hex(self.value)})"
+
+    def __hash__(self) -> int:
+        return hash((self.value, self.__class__))
+
+    def __rshift__(self, alias: str) -> tuple[ScType, str]:
+        return self, alias
+
+    def __eq__(self, other: ScType) -> bool:
+        if isinstance(other, ScType):
+            return self.value == other.value
+        return NotImplemented
+
+    def __bool__(self) -> bool:
+        return self.value != 0
+
+    def has_constancy(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_CONSTANCY_MASK) != 0
+
+    def is_node(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE) != 0
+
+    def is_connector(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_CONNECTOR) != 0
+
+    def is_edge(self) -> bool:
+        warnings.warn("ScType `is_edge` method is deprecated. Use `is_connector` method instead.", DeprecationWarning)
+        return self.is_connector()
+
+    def is_common_edge(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_COMMON_EDGE) != 0
+
+    def is_arc(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_ARC) != 0
+
+    def is_common_arc(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_COMMON_ARC) != 0
+
+    def is_membership_arc(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_MEMBERSHIP_ARC) != 0
+
+    def is_link(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_LINK) == bitmasks.SC_TYPE_NODE_LINK
+
+    def is_const(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_CONST) != 0
+
+    def is_var(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_VAR) != 0
+
+    def is_pos(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_POS_ARC) == bitmasks.SC_TYPE_POS_ARC
+
+    def is_neg(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NEG_ARC) == bitmasks.SC_TYPE_NEG_ARC
+
+    def is_fuz(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_FUZ_ARC) == bitmasks.SC_TYPE_FUZ_ARC
+
+    def is_perm(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_PERM_ARC) == bitmasks.SC_TYPE_PERM_ARC
+
+    def is_temp(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_TEMP_ARC) == bitmasks.SC_TYPE_TEMP_ARC
+
+    def is_actual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_ACTUAL_ARC) == bitmasks.SC_TYPE_ACTUAL_ARC
+
+    def is_inactual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_INACTUAL_ARC) == bitmasks.SC_TYPE_INACTUAL_ARC
```

Comment:
why are some methods comparing result with 0 and some are comparing result with type? also a lot of duplicated code, can be in some method that takes type and returns bool

Reply:
okay